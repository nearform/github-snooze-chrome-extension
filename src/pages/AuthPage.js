import React, { useReducer } from 'react'
import {
  FormControl,
  InputLabel,
  InputAdornment,
  IconButton,
  Box,
  Typography,
  Link,
  Alert,
  OutlinedInput
} from '@mui/material'
import { LoadingButton } from '@mui/lab'
import { Visibility, VisibilityOff } from '@mui/icons-material'
import { getUserByPat } from '../api/github'
import {
  clearLocalStorage,
  clearSyncStorage,
  writeToLocalStorage
} from '../api/chrome'
import DialogButton from '../components/DialogButton'
import StyledText from '../components/StyledText'
import { COLOR_PRIMARY, COLOR_SECONDARY } from '../constants'

function AuthPage({ isAuthenticated, pat }) {
  const [state, setState] = useReducer(
    (state, newState) => ({ ...state, ...newState }),
    {
      isLoading: false,
      token: pat || '',
      showToken: false,
      formErrorMessage: ''
    }
  )

  const handleChange = e => {
    const { value } = e.target
    setState({ token: value })
  }

  const handleShowToken = () => setState({ showToken: !state.showToken })

  const handleLogin = async () => {
    setState({ isLoading: true, formErrorMessage: '' })

    const { token } = state
    try {
      if (!token) {
        throw Error('A token must be specified.')
      }

      const userData = await getUserByPat(token)

      if (!userData.id) {
        throw Error('The token is not valid.')
      }

      await writeToLocalStorage({ pat: token, user: userData })

      window.location.reload()
    } catch (err) {
      setState({ formErrorMessage: err.message })
    } finally {
      setState({ isLoading: false })
    }
  }

  const handleLogout = async () => {
    await clearLocalStorage()
    window.location.reload()
  }

  const handleClearSyncStorage = async () => {
    await clearSyncStorage()
    window.location.reload()
  }

  const { formErrorMessage, isLoading, showToken, token } = state

  return (
    <>
      {formErrorMessage && (
        <Alert sx={{ mb: 1 }} severity="error">
          {formErrorMessage}
        </Alert>
      )}
      <Typography variant="body1" component="p">
        In order to use this Chrome Extension you have to generate a{' '}
        <Link href="https://github.com/settings/tokens/new" target="_blank">
          GitHub Personal Access Token (PAT)
        </Link>{' '}
        and insert it into the box below.
      </Typography>
      <Typography variant="subtitle1" component="sub">
        Remember to add the{' '}
        <StyledText color={COLOR_PRIMARY} backgroundColor={COLOR_SECONDARY}>
          repo
        </StyledText>{' '}
        and{' '}
        <StyledText color={COLOR_PRIMARY} backgroundColor={COLOR_SECONDARY}>
          read:user
        </StyledText>{' '}
        scopes to it.
      </Typography>
      <Box height={20} />
      <FormControl
        fullWidth
        color="secondary"
        variant="outlined"
        error={formErrorMessage !== ''}
        disabled={isLoading}
      >
        <InputLabel htmlFor="personal-access-token">PAT</InputLabel>
        <OutlinedInput
          id="personal-access-token"
          type={showToken ? 'text' : 'password'}
          value={token}
          onChange={handleChange}
          endAdornment={
            <InputAdornment position="end">
              <IconButton
                aria-label="toggle token visibility"
                onClick={handleShowToken}
                edge="end"
              >
                {showToken ? <VisibilityOff /> : <Visibility />}
              </IconButton>
            </InputAdornment>
          }
          label="PAT"
        />
        <Box height={20} />
        <LoadingButton
          fullWidth
          color="secondary"
          variant="contained"
          onClick={handleLogin}
          loading={isLoading}
          disabled={!token || token === pat}
        >
          {isAuthenticated ? 'Save' : 'Login'}
        </LoadingButton>
        {isAuthenticated && (
          <>
            <Box height={20} />
            <DialogButton
              label="Logout"
              title="Do you want to logout?"
              description="All your local data will be canceled and you won't be able to use the extension anymore until you log in again."
              onConfirm={handleLogout}
            />
            <Box height={20} />
            <DialogButton
              label="Clear Sync Storage"
              title="Do you want to clear the sync storage?"
              description="All your sync data (snooze list and badge counter) will be canceled across devices."
              onConfirm={handleClearSyncStorage}
            />
          </>
        )}
      </FormControl>
    </>
  )
}

export default AuthPage
