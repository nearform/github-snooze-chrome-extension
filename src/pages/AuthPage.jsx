import React, { useReducer } from 'react'
import {
  Alert,
  FormControl,
  FormControlLabel,
  IconButton,
  InputAdornment,
  InputLabel,
  Link,
  OutlinedInput,
  Radio,
  RadioGroup,
  Typography
} from '@mui/material'
import { LoadingButton } from '@mui/lab'
import { Visibility, VisibilityOff } from '@mui/icons-material'
import { getUserByPat } from '../api/github'
import {
  clearLocalStorage,
  clearSyncStorage,
  writeToLocalStorage
} from '../api/chrome'
import DialogButton from '../components/DialogButton'
import StyledText from '../components/StyledText'
import Gap from '../components/Gap'
import { TOKEN_TYPES } from '../constants'

function AuthPage({ isAuthenticated, pat }) {
  const [state, setState] = useReducer(
    (state, newState) => ({ ...state, ...newState }),
    {
      isLoading: false,
      token: pat || '',
      showToken: false,
      formErrorMessage: ''
    }
  )

  const [tokenType, setTokenType] = React.useState(TOKEN_TYPES.FINE_GRAINED)

  const handleChange = e => {
    const { value } = e.target
    setState({ token: value })
  }

  const handleShowToken = () => setState({ showToken: !state.showToken })

  const handleLogin = async () => {
    setState({ isLoading: true, formErrorMessage: '' })

    const { token } = state
    try {
      if (!token) {
        throw Error('A token must be specified.')
      }

      const userData = await getUserByPat(token)

      if (!userData.id) {
        throw Error('The token is not valid.')
      }

      await writeToLocalStorage({ pat: token, user: userData })

      window.location.reload()
    } catch (err) {
      setState({ formErrorMessage: err.message })
    } finally {
      setState({ isLoading: false })
    }
  }

  const handleLogout = async () => {
    await clearLocalStorage()
    window.location.reload()
  }

  const handleClearSyncStorage = async () => {
    await clearSyncStorage()
    window.location.reload()
  }

  const { formErrorMessage, isLoading, showToken, token } = state

  const onTokenTypeChange = (value) => {
    setTokenType(value)
  }

  return (
    <>
      {formErrorMessage && (
        <Alert sx={{ mb: 1 }} severity="error">
          {formErrorMessage}
        </Alert>
      )}
      <Typography variant="body1" component="p">
        In order to use this Chrome Extension you have to generate a Github Personal Access Token (PAT){' '}
        and insert it into the box below.
      </Typography>
      <Typography variant="body1" component="div">
        This can be done by generating one of the following PATs
        <RadioGroup
          row
          aria-labelledby="demo-row-radio-buttons-group-label"
          name="row-radio-buttons-group"
          value={tokenType}
          onChange={(event) => onTokenTypeChange(event.target.value)}
        >
          <FormControlLabel value={TOKEN_TYPES.FINE_GRAINED} control={<Radio color="default"/>} label="Fine-grained PAT" />
          <FormControlLabel value={TOKEN_TYPES.CLASSIC} control={<Radio color="default"/>} label="Classic PAT" />
        </RadioGroup>
      </Typography>

      <Typography fontWeight="bold" variant="body1" component="p">
        {tokenType === TOKEN_TYPES.FINE_GRAINED && <Link
              color="secondary"
              href="https://github.com/settings/personal-access-tokens/new"
              target="_blank"
            >
              Fine-grained PAT
          </Link>
        }
        {tokenType === TOKEN_TYPES.CLASSIC && <Link
              color="secondary"
              href="https://github.com/settings/tokens/new"
              target="_blank"
            >
              Classic PAT
          </Link>
        }
      </Typography>
      {tokenType === TOKEN_TYPES.FINE_GRAINED && <>
          <Typography variant="subtitle1" component="sub">
            In the Repository access section, select <StyledText as="span">All repositories</StyledText>.
          </Typography>
          <Typography component="p">
            Add the <StyledText as="span">Read-only</StyledText>{' '}
            permission to the following Repository permissions:
          </Typography>
          <Typography variant="subtitle1" component="sub">
            <StyledText as="span">Issues</StyledText> and{' '}
            <StyledText as="span">Pull requests</StyledText>
          </Typography>
        </>
      }
      {tokenType === TOKEN_TYPES.CLASSIC && <>
          <Typography variant="subtitle1" component="sub">
            Set the following scopes: <StyledText as="span">repo</StyledText> and{' '}
            <StyledText as="span">read:user</StyledText>.
          </Typography>
        </>
      }
      <Gap />
      <FormControl
        fullWidth
        color="secondary"
        variant="outlined"
        error={formErrorMessage !== ''}
        disabled={isLoading}
      >
        <InputLabel htmlFor="personal-access-token">PAT</InputLabel>
        <OutlinedInput
          id="personal-access-token"
          type={showToken ? 'text' : 'password'}
          value={token}
          onChange={handleChange}
          endAdornment={
            <InputAdornment position="end">
              <IconButton
                aria-label="toggle token visibility"
                onClick={handleShowToken}
                edge="end"
              >
                {showToken ? <VisibilityOff /> : <Visibility />}
              </IconButton>
            </InputAdornment>
          }
          label="PAT"
        />
        <Gap />
        <LoadingButton
          fullWidth
          color="secondary"
          variant="contained"
          onClick={handleLogin}
          loading={isLoading}
          disabled={!token || token === pat}
        >
          {isAuthenticated ? 'Save' : 'Login'}
        </LoadingButton>
        {isAuthenticated && (
          <>
            <Gap />
            <DialogButton
              label="Logout"
              title="Do you want to logout?"
              description="All your local data will be canceled and you won't be able to use the extension anymore until you log in again."
              onConfirm={handleLogout}
            />
            <Gap />
            <DialogButton
              label="Clear Sync Storage"
              title="Do you want to clear the sync storage?"
              description="All your sync data (snooze list and badge counter) will be canceled across devices."
              onConfirm={handleClearSyncStorage}
            />
          </>
        )}
      </FormControl>
    </>
  )
}

export default AuthPage
