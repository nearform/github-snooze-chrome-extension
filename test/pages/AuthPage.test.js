import React from 'react'
import renderer from 'react-test-renderer'
import AuthPage from '../../src/pages/AuthPage'

const props = {
  isAuthenticated: false,
  pat: '123'
}

describe('AuthPage.js', () => {
  test('renders the Auth Page for unauthenticated users', () => {
    const tree = renderer.create(<AuthPage {...props} />)

    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('renders the Auth Page for authenticated users', () => {
    const tree = renderer.create(<AuthPage {...props} isAuthenticated={true} />)

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
