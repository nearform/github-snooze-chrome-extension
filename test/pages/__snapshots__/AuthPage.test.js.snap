// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPage.js renders the Auth Page for authenticated users 1`] = `
Array [
  "",
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    In order to use this Chrome Extension you have to generate a
     
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
      href="https://github.com/settings/tokens/new"
      onBlur={[Function]}
      onFocus={[Function]}
      target="_blank"
    >
      GitHub Personal Access Token (PAT)
    </a>
     
    and insert it into the box below.
  </p>,
  <sub
    className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
  >
    Remember to add the
     
    <span
      style={
        Object {
          "backgroundColor": "#FB7A9C",
          "borderRadius": "25px",
          "color": "#140048",
          "padding": "3px",
        }
      }
    >
      repo
    </span>
     
    and
     
    <span
      style={
        Object {
          "backgroundColor": "#FB7A9C",
          "borderRadius": "25px",
          "color": "#140048",
          "padding": "3px",
        }
      }
    >
      read:user
    </span>
     
    scopes to it.
  </sub>,
  <div
    className="MuiBox-root css-1pg8bi"
  />,
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor="personal-access-token"
    >
      PAT
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="personal-access-token"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="password"
        value="123"
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="toggle token visibility"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1ftyaf0"
        >
          <span>
            PAT
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      className="MuiBox-root css-1pg8bi"
    />
    <button
      className="MuiButtonBase-root Mui-disabled MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1xy3nps-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      disabled={true}
      id="mui-2"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      Save
    </button>
    <div
      className="MuiBox-root css-1pg8bi"
    />
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth css-slst40-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Logout
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      className="MuiBox-root css-1pg8bi"
    />
    <div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth css-slst40-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Clear Sync Storage
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
]
`;

exports[`AuthPage.js renders the Auth Page for unauthenticated users 1`] = `
Array [
  "",
  <p
    className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    In order to use this Chrome Extension you have to generate a
     
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
      href="https://github.com/settings/tokens/new"
      onBlur={[Function]}
      onFocus={[Function]}
      target="_blank"
    >
      GitHub Personal Access Token (PAT)
    </a>
     
    and insert it into the box below.
  </p>,
  <sub
    className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
  >
    Remember to add the
     
    <span
      style={
        Object {
          "backgroundColor": "#FB7A9C",
          "borderRadius": "25px",
          "color": "#140048",
          "padding": "3px",
        }
      }
    >
      repo
    </span>
     
    and
     
    <span
      style={
        Object {
          "backgroundColor": "#FB7A9C",
          "borderRadius": "25px",
          "color": "#140048",
          "padding": "3px",
        }
      }
    >
      read:user
    </span>
     
    scopes to it.
  </sub>,
  <div
    className="MuiBox-root css-1pg8bi"
  />,
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor="personal-access-token"
    >
      PAT
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id="personal-access-token"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="password"
        value="123"
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="toggle token visibility"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
        </button>
      </div>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1in441m"
        >
          <span>
            PAT
          </span>
        </legend>
      </fieldset>
    </div>
    <div
      className="MuiBox-root css-1pg8bi"
    />
    <button
      className="MuiButtonBase-root Mui-disabled MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1xy3nps-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      Login
    </button>
  </div>,
]
`;
